### PR TITLE
Prompt for user when the config has a token but no user. Fixes #1007

### LINKED ISSUE #1007: `hub clone <repo>` does not work at hub version 2.2.1
`hub clone <owner>/<repo>` does work.
the feature was changed?

```
wkentaro at anaconda in tmp
$ /usr/local/bin/hub --version
git version 1.9.1
hub version 1.12.4
wkentaro at anaconda in tmp
$ /usr/local/bin/hub clone dotfiles hoge
Cloning into 'hoge'...
remote: Counting objects: 3837, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 3837 (delta 0), reused 0 (delta 0), pack-reused 3833
Receiving objects: 100% (3837/3837), 1.15 MiB | 309.00 KiB/s, done.
Resolving deltas: 100% (1992/1992), done.
Checking connectivity... done.
wkentaro at anaconda in tmp
$ hub --version
git version 1.9.1
hub version 2.2.1
wkentaro at anaconda in tmp
$ hub clone dotfiles fuga
Cloning into 'fuga'...
fatal: remote error: 
  Repository not found.
wkentaro at anaconda in tmp
$ hub clone wkentaro/dotfiles fuga
Cloning into 'fuga'...
remote: Counting objects: 3837, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 3837 (delta 0), reused 0 (delta 0), pack-reused 3833
Receiving objects: 100% (3837/3837), 1.15 MiB | 326.00 KiB/s, done.
Resolving deltas: 100% (1992/1992), done.
Checking connectivity... done.
```

### PR DESCRIPTION
**Pull request by pcorpet**

### CODE REVIEW

File: features/steps.rb [hunk 1]
```diff
@@ -281,3 +281,10 @@
     File.open(file, "w") { |f| f.write(contents) }
   end
 end
+
+Given(/^the file "([^"]+)" content is:$/) do |file, content|
```
Comment (mislav) [inline features/steps.rb]: There's no need: you can simply re-use the `Given a file named "..." with:` step provided by Aruba.

File: github/config.go [hunk 2]
```diff
@@ -48,6 +48,14 @@ func (c *Config) PromptForHost(host string) (h *Host, err error) {
 
 	h = c.Find(host)
 	if h != nil {
+		if h.User == "" {
+			// User is missing from the config: this is a broken config probably
+			// because it was created with an old (broken) version of hub. Let's fix
+			// it now. See issue #1007 for details.
+			h.User = c.PromptForUser(host)
```
Comment (mislav) [inline github/config.go]: Should we check if the input was blank?

**Comment by mislav** [comment]
Protip: when you reference a fixed issue, put `Fixes #XYZ` in the pull request body. That will make the issue auto-close when the PR is merged.

File: features/steps.rb (hunk 1 above)
Comment (pcorpet) [inline features/steps.rb]: Thanks, I didn't know about that one. I removed that code.

File: github/config.go (hunk 2 above)
Comment (pcorpet) [inline github/config.go]: Done. I added a scenario for this too.

**Comment by pcorpet** [comment]
Thanks for the protip. I had it in the commit (which already gives a hint in the issue that someone's working on it). And it's in the PR title (instead of body), I do think it's enough and that the issue will be closed when the PR is merged.

**Comment by mislav** [comment]
> And it's in the PR title (instead of body), I do think it's enough and that the issue will be closed when the PR is merged.

No, issue referenced from the PR title don't count as actual references and won't auto-close the issue.

**Comment by mislav** [comment]
Oh, but because you had the issue reference within commit message as well, all is good since the issue got auto-closed. :+1: Thanks!

**Comment by pcorpet** [comment]
> No, issue referenced from the PR title don't count as actual references and won't auto-close the issue.

The one in the commit was enough.

**Pull request merged**